Some integration tests fail to build

Running `cargo check --tests --features docker` results in a number of errors:

```
    Checking vector v0.11.0 (/home/bruce/Timber/vector)
error[E0599]: no method named `compat` found for struct `std::pin::Pin<std::boxed::Box<dyn core::future::future::Future<Output = std::result::Result<(), std::boxed::Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>>> + std::marker::Send>>` in the current scope
   --> src/sinks/gcp/pubsub.rs:256:21
    |
256 |           healthcheck.compat().await.expect("Health check failed");
    |                       ^^^^^^ method not found in `std::pin::Pin<std::boxed::Box<dyn core::future::future::Future<Output = std::result::Result<(), std::boxed::Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>>> + std::marker::Send>>`
    | 
   ::: /home/bruce/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.5/src/future/try_future/mod.rs:571:8
```

`git bisect` indicates the failures were caused by 2272cca0f15812982d81e88ea516850ca45707e6 (#3685).

--- a/src/sinks/aws_cloudwatch_logs/mod.rs
+++ b/src/sinks/aws_cloudwatch_logs/mod.rs
@@ -885,7 +885,7 @@ mod integration_tests {
         sink.run(events).await.unwrap();
 
         let mut request = GetLogEventsRequest::default();
-        request.log_stream_name = stream_name.clone().into();
+        request.log_stream_name = stream_name;
         request.log_group_name = GROUP_NAME.into();
         request.start_time = Some(timestamp.timestamp_millis());
 
@@ -945,7 +945,7 @@ mod integration_tests {
         let _ = sink.run(events).await.unwrap();
 
         let mut request = GetLogEventsRequest::default();
-        request.log_stream_name = stream_name.clone().into();
+        request.log_stream_name = stream_name;
         request.log_group_name = GROUP_NAME.into();
         request.start_time = Some(timestamp.timestamp_millis());
 
@@ -1017,7 +1017,7 @@ mod integration_tests {
         sink.run(stream::iter(events)).await.unwrap();
 
         let mut request = GetLogEventsRequest::default();
-        request.log_stream_name = stream_name.clone().into();
+        request.log_stream_name = stream_name;
         request.log_group_name = GROUP_NAME.into();
         request.start_time = Some((now - Duration::days(30)).timestamp_millis());
 
@@ -1061,7 +1061,7 @@ mod integration_tests {
         sink.run(events).await.unwrap();
 
         let mut request = GetLogEventsRequest::default();
-        request.log_stream_name = stream_name.clone().into();
+        request.log_stream_name = stream_name;
         request.log_group_name = group_name;
         request.start_time = Some(timestamp.timestamp_millis());
 
@@ -1116,8 +1116,8 @@ mod integration_tests {
             .unwrap();
 
         let mut request = GetLogEventsRequest::default();
-        request.log_stream_name = stream_name.clone().into();
-        request.log_group_name = group_name.into();
+        request.log_stream_name = stream_name;
+        request.log_group_name = group_name;
         request.start_time = Some(timestamp.timestamp_millis());
 
         let response = create_client_test().get_log_events(request).await.unwrap();

--- a/src/sinks/aws_s3.rs
+++ b/src/sinks/aws_s3.rs
@@ -722,7 +722,7 @@ mod integration_tests {
     }
 
     async fn get_keys(prefix: String) -> Vec<String> {
-        let prefix = prefix.split("/").into_iter().next().unwrap().to_string();
+        let prefix = prefix.split('/').next().unwrap().to_string();
 
         let list_res = client()
             .list_objects_v2(rusoto_s3::ListObjectsV2Request {

--- a/src/sinks/elasticsearch.rs
+++ b/src/sinks/elasticsearch.rs
@@ -597,7 +597,7 @@ mod integration_tests {
 
         let config = ElasticSearchConfig {
             endpoint: "http://localhost:9200".into(),
-            index: Some(index.clone()),
+            index: Some(index),
             pipeline: Some(pipeline.clone()),
             ..config()
         };
@@ -742,24 +742,21 @@ mod integration_tests {
         healthcheck.await.expect("Health check failed");
 
         let (input, events) = random_events_with_stream(100, 100);
-        match break_events {
-            true => {
-                // Break all but the first event to simulate some kind of partial failure
-                let mut doit = false;
-                sink.run(events.map(move |mut event| {
-                    if doit {
-                        event.as_mut_log().insert("_type", 1);
-                    }
-                    doit = true;
-                    event
-                }))
-                .await
-                .expect("Sending events failed");
-            }
-            false => {
-                sink.run(events).await.expect("Sending events failed");
-            }
-        };
+        if break_events {
+            // Break all but the first event to simulate some kind of partial failure
+            let mut doit = false;
+            sink.run(events.map(move |mut event| {
+                if doit {
+                    event.as_mut_log().insert("_type", 1);
+                }
+                doit = true;
+                event
+            }))
+            .await
+            .expect("Sending events failed");
+        } else {
+            sink.run(events).await.expect("Sending events failed");
+        }
 
         // make sure writes all all visible
         flush(cx.resolver(), common)

--- a/src/sinks/gcp/pubsub.rs
+++ b/src/sinks/gcp/pubsub.rs
@@ -305,7 +305,7 @@ mod integration_tests {
             .json(&json)
             .send()
             .await
-            .expect(&format!("Sending {} request to {} failed", method, url))
+            .unwrap_or_else(|_| panic!("Sending {} request to {} failed", method, url))
     }
 
     async fn pull_messages(subscription: &str, count: usize) -> PullResponse {

--- a/src/sinks/humio_logs.rs
+++ b/src/sinks/humio_logs.rs
@@ -175,7 +175,9 @@ mod integration_tests {
         assert!(
             entry.error.is_none(),
             "Humio encountered an error parsing this message: {}",
-            entry.error_msg.unwrap_or("no error message".to_string())
+            entry
+                .error_msg
+                .unwrap_or_else(|| "no error message".to_string())
         );
     }
 
@@ -200,7 +202,9 @@ mod integration_tests {
         assert!(
             entry.error.is_none(),
             "Humio encountered an error parsing this message: {}",
-            entry.error_msg.unwrap_or("no error message".to_string())
+            entry
+                .error_msg
+                .unwrap_or_else(|| "no error message".to_string())
         );
     }
 
@@ -231,7 +235,9 @@ mod integration_tests {
             assert!(
                 entry.error.is_none(),
                 "Humio encountered an error parsing this message: {}",
-                entry.error_msg.unwrap_or("no error message".to_string())
+                entry
+                    .error_msg
+                    .unwrap_or_else(|| "no error message".to_string())
             );
         }
 

--- a/src/sinks/influxdb/logs.rs
+++ b/src/sinks/influxdb/logs.rs
@@ -694,10 +694,10 @@ mod integration_tests {
             .unwrap();
         let string = res.text().await.unwrap();
 
-        let lines = string.split("\n").collect::<Vec<&str>>();
-        let header = lines[0].split(",").collect::<Vec<&str>>();
-        let record1 = lines[1].split(",").collect::<Vec<&str>>();
-        let record2 = lines[2].split(",").collect::<Vec<&str>>();
+        let lines = string.split('\n').collect::<Vec<&str>>();
+        let header = lines[0].split(',').collect::<Vec<&str>>();
+        let record1 = lines[1].split(',').collect::<Vec<&str>>();
+        let record2 = lines[2].split(',').collect::<Vec<&str>>();
 
         // measurement
         assert_eq!(

--- a/src/sinks/influxdb/metrics.rs
+++ b/src/sinks/influxdb/metrics.rs
@@ -899,9 +899,9 @@ mod integration_tests {
             .unwrap();
         let string = res.text().await.unwrap();
 
-        let lines = string.split("\n").collect::<Vec<&str>>();
-        let header = lines[0].split(",").collect::<Vec<&str>>();
-        let record = lines[1].split(",").collect::<Vec<&str>>();
+        let lines = string.split('\n').collect::<Vec<&str>>();
+        let header = lines[0].split(',').collect::<Vec<&str>>();
+        let record = lines[1].split(',').collect::<Vec<&str>>();
 
         assert_eq!(
             record[header

--- a/src/sinks/loki.rs
+++ b/src/sinks/loki.rs
@@ -273,7 +273,7 @@ mod tests {
     }
 }
 
-#[cfg(feature = "docker")]
+#[cfg(feature = "loki-integration-tests")]
 #[cfg(test)]
 mod integration_tests {
     use super::*;
@@ -310,6 +310,7 @@ mod integration_tests {
 
         let events = lines.clone().into_iter().map(Event::from);
         let _ = sink
+            .into_futures01sink()
             .send_all(futures01::stream::iter_ok(events))
             .compat()
             .await
@@ -347,6 +348,7 @@ mod integration_tests {
             .map(Event::from)
             .collect::<Vec<_>>();
         let _ = sink
+            .into_futures01sink()
             .send_all(futures01::stream::iter_ok(events.clone()))
             .compat()
             .await
@@ -394,6 +396,7 @@ mod integration_tests {
         }
 
         let _ = sink
+            .into_futures01sink()
             .send_all(futures01::stream::iter_ok(events))
             .compat()
             .await

